app/routes.py

```diff
@@ -1,2 +1,197 @@
+    else:
+        tasks = Task.query.all()
+    
+    tasks_response = []
+    for task in tasks:
+        # building a list of jsons by calling to_json on each task
+        tasks_response.append(task.to_json())
+
+    return jsonify(tasks_response)
+
+
+@task_list_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+    elif request.method == "GET":
+        if task.goal_offspring is None:
+            return make_response({"task": task.to_json()}, 200)
+        else:
+            return make_response({"task": task.goal_json()}, 200)
+    
+    elif request.method == "PUT":
+        task_data = request.get_json()
+        task.title = task_data["title"]
+        task.description = task_data["description"]
+
+        db.session.commit()
+        
+        return make_response({"task": task.to_json()}, 200)
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return make_response({"details": f'Task {task.task_id} "{task.title}" successfully deleted'})
+
+
+@task_list_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    # Consider using the keyword arguments data, json, and/or headers
+    if task is None:
+        return make_response("", 404)
+    else:
+        task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    url = "https://slack.com/api/chat.postMessage"
+    data = {
+        "channel": "C021ACT6H0B",
+        "text": (f"Someone just completed the task {task.title}")
+    }
+    headers = {
+        "Authorization": f"Bearer {slack_key}"
+    }
+
+    r = requests.post(url, data=data, headers=headers)
+
+    return make_response({"task": task.to_json()})
+
+
+@task_list_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+    else:
+        task.completed_at = None
+        db.session.commit()
+
+    return make_response({"task": task.to_json()})
+
+
+# WAVE 5 & 6 GOALS
+
+@goal_list_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    else:
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
+        db.session.commit() 
+    
+    # returning single task created by calling to_json on new task
+    return make_response({"goal": new_goal.to_json()}, 201)
+    
+
+@goal_list_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        # building a list of jsons by calling to_json on each task
+        goals_response.append(goal.to_json())
+
+    return jsonify(goals_response, 200)
```

Comment:
Not a big thing but to explain: The two tests are failing for this because `jsonify` doesn't expect a status code the way `make_response` does. The code I have below would fix the problem.
```suggestion
    return jsonify(goals_response), 200
```

Reply:
This would also work
```suggestion
    return make_response(jsonify(goals_response), 200)
```